### PR TITLE
Cython API refactor: `merge.pyx`

### PR DESCRIPTION
**Pull request by isVoid**
This PR refactors `merge_sorted` in `merge.pyx` to accept a list of columns, contributes to #10153

### CODE REVIEW

File: python/cudf/cudf/_lib/merge.pyx [hunk 1]
```diff
@@ -10,79 +10,39 @@ from cudf._lib.column cimport Column
...
-from cudf._lib.utils cimport data_from_unique_ptr, table_view_from_table
+from cudf._lib.utils cimport columns_from_unique_ptr, table_view_from_columns
 
 
 def merge_sorted(
-    object tables,
-    object keys=None,
-    bool by_index=False,
-    bool ignore_index=False,
+    list list_of_columns,
```
Comment (shwina) [inline python/cudf/cudf/_lib/merge.pyx:17]: Perhaps

```suggestion
    list input_columns,
```

?

**Review by shwina — APPROVED** [review]
Minor nit

File: python/cudf/cudf/_lib/merge.pyx (hunk 1 above)
Comment (isVoid) [reply python/cudf/cudf/_lib/merge.pyx]: Actually, I'm trying to differentiate this with other cython APIs where we are expecting `List[List[Column]]` here, not just `List[Column]`. I'm still fond of `input_columns` but I'll add an informal docstring to describe this better.

**Comment by isVoid** [comment]
rerun tests

**Comment by isVoid** [comment]
@gpucibot merge

**Pull request merged**